FirebaseDeviceStore/FirebaseDeviceStore.swift

```diff
@@ -34,6 +34,8 @@ public class FirebaseDeviceStore: NSObject, MessagingDelegate {
         self.instanceId = InstanceID.instanceID();
 
         super.init();
+        // TODO: Is this going to prevent the developer from setting a Messaging delegate (and
+        // therefore from implementing didReceiveMessage, etc.)?
```

Comment:
This is a very good point that I hadn't considered.

I guess we could provide this as a configuration:
- if the developer doesn't have their own delegate then allow it to be set by the library
- Otherwise, expose the delegate method and ask the user to call it as part of their own delegate

Unless you have any other suggestions?

It's a shame that the iOS SDK doesn't support multiple delegates.  I don't think this is an issue on Android as you can register multiple `FirebaseMessagingService` implementations in the `AndroidManifest`.

Reply:
Offhand I don't know of any better solution, but I've tried to ping some folks internally to see if anybody else does.

Reply:
Please file a feature request for register multiple delegate. Otherwise, you might have to merge this class with AppDelegate to share one delegate at this point.

Reply:
@chliangGoogle Thanks for the suggestion, I've adopted that approach